submissions/andrewklmn/friends-app/app.js

```diff
@@ -0,0 +1,302 @@
+
+const maxNumberOfFriends = 500;
+const randomUserUrl = 'https://randomuser.me/api/?results=' + maxNumberOfFriends + '&seed=friends';
+
+const state = {
+  friends: [],
+  filters: {
+    partOfName: '',
+    gender: '',
+    genderList: [],
+    ageRange: [18,99],
+    country: '',
+    countryList: [],
+  },
+  sorter: {
+    keyName: '',
+    order: '',
+    fieldPathList: {
+      name: ['name', 'first'],
+      age: ['dob', 'age'],
+      country: ['location', 'country'],
+    },
+    orderSymbols: {
+      ASC: '&#9650;', 
+      DESC: '&#9660;',
+    },
+  },
+  initialListLength: 25,
+  numberOfShowedFriends: 0,
+  scrollDisabled: false,
+  nextMoreFriendAutoloadDelay: 300,
+}
+
+const preloader = document.querySelector('.preloader');
+const container = document.querySelector('.container');
+
+const sortField = document.querySelector('.sort-field');
+const sortOrder = document.querySelector('.sort-order');
+
+const searchField = document.querySelector('.search-field');
+const filterGender = document.querySelector('.filter-gender');
+const filterMinAge = document.querySelector('.filter-min-age');
+const filterMaxAge = document.querySelector('.filter-max-age');
+const filterCountry = document.querySelector('.filter-country');
+
+const drawPerson = (person) => {  
+  const div = document.createElement('div');
+  div.classList.add('person');
+  div.title = `Address: ${person.location.street.number}, ${person.location.street.name}, ${person.location.city}, ${person.location.state}
+  Phone: ${person.phone}
+  Email: ${person.email}`;
+  div.innerHTML =`
+      <div class="person-name ${person.gender}">${person.name.first} ${person.name.last}</div>
+      <div class="person-image">
+      <picture>
+        <source media="(max-width:300px)" 
+                srcset="${person.picture.thumbnail} 1x, ${person.picture.medium} 2x">
+        <source media="(max-width:400px)" 
+                srcset="${person.picture.medium} 1x, ${person.picture.large} 2x">
+        <img class="rounded" src="${person.picture.large}">
+      </picture>
+      </div>
+      <div class="person-age">
+        ${person.dob.age} y.o.
+      </div>
+      <div class="person-location">
+        ${person.location.country}
+      </div>
+  `;
+  container.appendChild(div);
+}
+
+const filterList = (friends)=>{
+  return friends.filter( person => {
+    if (state.filters.namePart != '') {
+      if((person.name.first + ' ' + person.name.last).toUpperCase()
```

Comment:
Use destructuring and [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)